src/core/Akka/Actor/Stash/Internal/AbstractStash.cs

```diff
@@ -61,12 +63,13 @@ public void Stash()
         {
             var currMsg = _actorCell.CurrentMessage;
             var sender = _actorCell.Sender;
-            if(_theStash.Count > 0)
+
+            if (_actorCell.CurrentEnvelopeId == _currentEnvelopeId)
```

Comment:
This solution could fail if you receive exactly 4 294 967 294 messages since your last Stash call.. let it crash?